packages/blade/src/components/ActionList/ActionList.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import Box from '~components/Box';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { getPlatformType, makeAccessible } from '~utils';
+
+type ActionListItemProps = {
+  title: string;
+  value: string;
+  href?: string;
+  /**
+   * Internally passed from ActionList. No need to pass it explicitly
+   */
+  index?: number;
+};
+const ActionListItem = (props: ActionListItemProps): JSX.Element => {
+  const {
+    activeIndex,
+    dropdownBaseId,
+    onOptionClick,
+    selectedIndices,
+    setShouldIgnoreBlur,
+    selectInputRef,
+  } = useDropdown();
```

Comment:
if we use these as dropdown then this is very much tied to dropdown right? we can't use this action list for other things like normal listbox or menu in navbars?

Reply:
So atleast for now we don't have a usecase but it might come up in future.

There are several questions we have to tackle when we do that. Like what happens to the selection logic or will multiple selection still work there, etc.

Right now I am leaning more towards an opinion of having a different Navbar/MenuItems component in itself. We might have better clarity based on Layout components. If Layout component itself makes it easy to build those menus then not sure if we'll need it.

Overall I feel things will get too complicated if try to keep it generic for non-dropdown things. It might also impact bundle size where a normal anchor tag in Navbar will start loading all the dropdown logic.

Reply:
I have anyway kept most dropdown logic in `useDropdown` for now. So even if we plan to do that, it will just be few if else checks